src/tdlibreceiver.cpp

```diff
@@ -164,123 +212,139 @@ void TDLibReceiver::processUpdateUnreadChatCount(const QVariantMap &receivedInfo
     chatCountInformation.insert("total_count", receivedInformation.value("total_count"));
     chatCountInformation.insert("unread_count", receivedInformation.value("unread_count"));
     chatCountInformation.insert("unread_unmuted_count", receivedInformation.value("unread_unmuted_count"));
-    qDebug() << "[TDLibReceiver] Unread chat count updated: " << chatCountInformation.value("chat_list_type").toString() << chatCountInformation.value("unread_count").toString();
+    LOG("Unread chat count updated: " << chatCountInformation.value("chat_list_type").toString() << chatCountInformation.value("unread_count").toString());
     emit unreadChatCountUpdated(chatCountInformation);
 }
 
 void TDLibReceiver::processUpdateChatLastMessage(const QVariantMap &receivedInformation)
 {
-    QVariantMap lastMessage = receivedInformation.value("last_message").toMap();
-    qDebug() << "[TDLibReceiver] Last message of chat " << receivedInformation.value("chat_id").toString() << " updated, order " << receivedInformation.value("order").toString() << " content: " << lastMessage.value("content").toString();
-    emit chatLastMessageUpdated(receivedInformation.value("chat_id").toString(), receivedInformation.value("order").toString(), lastMessage);
+    const QString chat_id(receivedInformation.value(CHAT_ID).toString());
+    const QString order(findChatPositionOrder(receivedInformation.value(POSITIONS).toList()));
```

Comment:
This doesn't work with TDLib 1.6.0 - the change with 'positions' and 'updateChatPosition' was introduced in May with https://github.com/tdlib/td/commit/06136146511a46bcc26fcb936e3e9ae42130e92c, so after 1.6.0. However, a small if/else seems to do the trick - so no blocker. :)

Reply:
Good to know, I'll switch my build to 1.6.0 too.